aws/adapter.go

```diff
@@ -119,6 +127,10 @@ func NewAdapter() (adapter *Adapter, err error) {
 		healthCheckInterval: DefaultHealthCheckInterval,
 		creationTimeout:     DefaultCreationTimeout,
 		stackTTL:            DefaultStackTTL,
+		autoScalingGroups:   make(map[string]*autoScalingGroupDetails),
+		instancesDetails:    make(map[string]*instanceDetails),
```

Comment:
instancesDetails is hard to read and spot the differences from instanceDetails type.
Maybe ec2Details?